nectarchain/calibration/container/charge.py

```diff
@@ -0,0 +1,346 @@
+from argparse import ArgumentError
+import numpy as np
+import numpy.ma as ma
+from matplotlib import pyplot as plt
+import copy
+from pathlib import Path
+import sys
+import os
+import logging
+logging.basicConfig(format='%(asctime)s %(name)s %(levelname)s %(message)s')
+log = logging.getLogger(__name__)
+log.handlers = logging.getLogger('__main__').handlers
+
+from ctapipe.visualization import CameraDisplay
+from ctapipe.coordinates import CameraFrame,EngineeringCameraFrame
+from ctapipe.instrument import CameraGeometry
+from ctapipe.image.extractor import (FullWaveformSum,
+    FixedWindowSum,
+    GlobalPeakWindowSum,
+    LocalPeakWindowSum,
+    SlidingWindowMaxSum,
+    NeighborPeakWindowSum,
+    BaselineSubtractedNeighborPeakWindowSum,
+    TwoPassWindowSum)
+
+from ctapipe_io_nectarcam import NectarCAMEventSource
+from ctapipe.io import EventSource, EventSeeker
+
+from astropy.table import Table
+from astropy.io import fits
+
+from .waveforms import WaveformsContainer,WaveformsContainers
+
+#from .charge_extractor import *
+
+__all__ = ['ChargeContainer','ChargeContainers']
+
+list_ctapipe_charge_extractor = ["FullWaveformSum",
+                        "FixedWindowSum",
+                        "GlobalPeakWindowSum",
+                        "LocalPeakWindowSum",
+                        "SlidingWindowMaxSum",
+                        "NeighborPeakWindowSum",
+                        "BaselineSubtractedNeighborPeakWindowSum",
+                        "TwoPassWindowSum"]
+
+list_nectarchain_charge_extractor = ['gradient_extractor']
+
+
+
+
+class ChargeContainer() : 
+    """class used to compute charge from waveforms container"""
+    TEL_ID = 0
+    CAMERA = CameraGeometry.from_name("NectarCam-003")
+
+    def __init__(self,charge_hg,charge_lg,peak_hg,peak_lg,run_number,pixels_id,nevents,npixels, method = "FullWaveformSum") : 
+        self.charge_hg = charge_hg
+        self.charge_lg = charge_lg
+        self.peak_hg = peak_hg
+        self.peak_lg = peak_lg
+        self.__run_number = run_number
+        self.__pixels_id = pixels_id
+        self.__method = method
+        self.__nevents = nevents
+        self.__npixels = npixels
+
+
+        self.ucts_timestamp = np.zeros((self.nevents),dtype = np.uint64)
+        self.ucts_busy_counter = np.zeros((self.nevents),dtype = np.uint16)
+        self.ucts_event_counter = np.zeros((self.nevents),dtype = np.uint16)
+        self.event_type = np.zeros((self.nevents),dtype = np.uint8)
+        self.event_id = np.zeros((self.nevents),dtype = np.uint16)
+        self.trig_pattern_all = np.zeros((self.nevents,self.CAMERA.n_pixels,4),dtype = bool)
+
+    @classmethod
+    def from_waveforms(cls,waveformContainer : WaveformsContainer,method : str = "FullWaveformSum",**kwargs) : 
+        log.info(f"computing hg charge with {method} method")
+        charge_hg,peak_hg = ChargeContainer.compute_charge(waveformContainer,0,method,**kwargs)
```

Comment:
Instead of hard-coding the index value for low/high gain channels, I would recommend to rely instead on https://github.com/cta-observatory/ctapipe_io_nectarcam/blob/master/ctapipe_io_nectarcam/constants.py